libs/SalesforceSDK/src/com/salesforce/androidsdk/ui/LoginActivity.kt

```diff
@@ -995,52 +975,39 @@ open class LoginActivity : FragmentActivity() {
         )
         .build()
 
-    /**
-     * Determines if the provided proposed selected server URL is a switch from
-     * Salesforce Welcome Discovery back to default log in.
-     * @param proposedSelectedServerUrl The proposed selected server URL
-     * @return Boolean true if the provided proposed selected server URL is a
-     * switch from Salesforce Welcome Discovery back to the default log in,
-     * false otherwise.
-     * */
-    private fun isSwitchFromSalesforceWelcomeDiscoveryToDefaultLogin(
-        proposedSelectedServerUrl: Uri
-    ) = viewModel.loginUrl.value?.toUri()?.let { loginUrl ->
-        isSalesforceWelcomeDiscoveryMobileUrl(loginUrl) && !(isSalesforceWelcomeDiscoveryMobileUrl(proposedSelectedServerUrl))
-    } ?: false
-
     /**
      * Switches between default or Salesforce Welcome Discovery log in as needed
-     * using the provided proposed selected server URL.
-     * @param uri The proposed selected server URL
+     * using the provided pending login server URL.
+     * @param pendingLoginServerUri The pending login server URL
      * @return Boolean true if a switch between default or Salesforce Welcome
      * Discovery log is made, false otherwise.
      */
-    private fun switchDefaultOrSalesforceWelcomeDiscoveryLogin(uri: Uri) =
+    @VisibleForTesting
+    internal fun switchDefaultOrSalesforceWelcomeDiscoveryLogin(pendingLoginServerUri: Uri) =
```

Comment:
What does `switchDefaultOrSalesforceWelcomeDiscoveryLogin` mean?  Maybe something like `useWelcomeDiscoveryIfNeeded` would be better?